rest/attachment_migration_test.go

```diff
@@ -0,0 +1,455 @@
+//  Copyright 2024-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	sgbucket "github.com/couchbase/sg-bucket"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestMigrationJobStartOnDbStart:
+//   - Create a db
+//   - Grab attachment migration manager and assert it has run upon db startup
+//   - Assert job has written syncInfo metaVersion as expected to the bucket
+func TestMigrationJobStartOnDbStart(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("rosmar does not support DCP client, pending CBG-4249")
+	}
+	rt := NewRestTesterPersistentConfig(t)
+	defer rt.Close()
+	ctx := base.TestCtx(t)
+
+	ds := rt.GetSingleDataStore()
+	dbCtx := rt.GetDatabase()
+
+	mgr := dbCtx.AttachmentMigrationManager
+
+	// wait for migration job to finish
+	db.RequireBackgroundManagerState(t, ctx, mgr, db.BackgroundProcessStateCompleted)
+
+	// assert that sync info with metadata version written to the collection
+	syncInf := base.SyncInfo{}
+	_, err := ds.Get(base.SGSyncInfo, &syncInf)
+	require.NoError(t, err)
+	assert.Equal(t, "4.0", syncInf.MetaDataVersion)
```

Comment:
This code is going to fail as soon as we bump to 4.1, watch where you use 4.0 as a string that isn't for test purposes. I'd be inclined _not_ to use 4.0 as an explicit string anywhere except for where you want to do a major version test.